### PR TITLE
refractored code

### PR DESCRIPTION
**Pull request by markroxor**
I have refactored the code so that it can - 

- [x] use `coverage` by considering the content of this repository as a package.
- [x] in future we can use pip to install this repository more conveniently.
- [x] the existing directory structure did not make so much sense. Eg. datasets belonged to models folder.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,8 +13,7 @@ __pycache__/
 .idea/
 
 # Files generated in experiments
-*.zip
```
Comment (quinnliu) [inline .gitignore]: okay thanks for removing this. we do want .zip files in this repo

File: walnut/tests/datasets/reading_dataset_tests.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-from nose.tools import *
```
Comment (quinnliu) [inline walnut/tests/datasets/reading_dataset_tests.py]: this is good :)

File: walnut/tests/experiments/classify_digits/classify_digits_tests.py [hunk 3]
```diff
@@ -1,12 +1,11 @@
-from nose.tools import *
 from zipfile import ZipFile
 from PIL import Image
-from model.retina import Retina
-from model.layer import Layer
-from model.network import Network
-from model.common_cortical_algorithm_v1 import CommonCorticalAlgorithmV1
-from model.connect_types import ConnectTypes
```
Comment (quinnliu) [inline walnut/tests/experiments/classify_digits/classify_digits_tests.py]: I think model.htmmodel seems redundant

File: walnut/model/htmmodel/connect_types.py [hunk 4]
```diff
@@ -45,7 +45,7 @@ def __update_receptive_field_dimension_length(top_length, bot_length, top_index,
 
         without_overlap = (b_initial, b_final)
```
Comment (markroxor) [inline walnut/model/htmmodel/connect_types.py]: In future we might have other models as well, HTW generative model for instance.

File: walnut/tests/experiments/classify_digits/classify_digits_tests.py (hunk 3 above)
Comment (markroxor) [inline walnut/tests/experiments/classify_digits/classify_digits_tests.py]: Models folder will hold all the models, HTM/HTW or whatever models we can think of in the future. Better create a base before hand.

File: walnut/model/htmmodel/connect_types.py (hunk 4 above)
Comment (quinnliu) [inline walnut/model/htmmodel/connect_types.py]: okay that makes sense

File: walnut/tests/experiments/classify_digits/classify_digits_tests.py (hunk 3 above)
Comment (quinnliu) [inline walnut/tests/experiments/classify_digits/classify_digits_tests.py]: okay good thinking

**Pull request merged**